### PR TITLE
allow encode from memoryview

### PR DESCRIPTION
**Pull request by mayli**
#69

### CODE REVIEW

File: src/sabyenc3.cc [hunk 1]
```diff
@@ -583,10 +583,17 @@ PyObject* encode(PyObject* self, PyObject* Py_input_string)
...
-        PyErr_SetString(PyExc_TypeError, "Expected bytes");
-        return NULL;
+    if(PyBytes_Check(Py_input_string)) {
+        input_len = PyBytes_Size(Py_input_string);
+        input_buffer = (char *)PyBytes_AsString(Py_input_string);
+    }
+    if(PyMemoryView_Check(Py_input_string)) {
+       Py_buffer *buffer = PyMemoryView_GET_BUFFER(Py_input_string);
+        input_len = buffer->len;
+        input_buffer = (char *)buffer->buf;
```
Comment (Safihre) [inline src/sabyenc3.cc:593]: Just a few lines down these variables get overwritten. I think some code is missing?

Comment (mayli) [reply src/sabyenc3.cc:593]: Fixed.

**Comment by Safihre** [comment]
Could you also add unittests?
For all types of input and expected (failure) output.

**Comment by Safihre** [comment]
Superseded by the recent efforts, new release will come soon.

**Pull request closed without merging**